src/lib/services/ws/ws.ts

```diff
@@ -0,0 +1,17 @@
+import type { Envelope } from '@lib/envelope/envelope';
+import type { ConnectMsg, MIDIMsg, TextMsg } from '@lib/types/jam';
+
+// planning to move everything inside classes
+class WSConn {
+    conn: WebSocket;
```

Comment:
That's cool with me, and maybe let's only expose methods to interact with the connection and not the connection itself
```suggestion
    private conn: WebSocket;
```